### PR TITLE
Critical health UI feedback

### PR DESCRIPTION
**Pull request by GBMiro**
When the player has less than 3 points of health, a vignette effect is displayed to give feedback to the player.

You can test it with the CriticalHealthWarning scene using the "DebugHit" checkbox in the Player script.

LowHP

### CODE REVIEW

**Comment by jcarlos0305** [comment]
Task linked: [CU-kk6zfz VS3 UI Features](https://app.clickup.com/t/kk6zfz)

**Comment by jcarlos0305** [comment]
Task linked: [CU-kk7483 Low hp vignetting](https://app.clickup.com/t/kk7483)

**Review by Jordi-Gil — APPROVED** [review]

File: Penteract/Assets/Scripts/HUDController.h [hunk 1]
```diff
@@ -96,6 +100,7 @@ class HUDController : public Script {
 
 	static int prevLives;
 	static float remainingTimes[];
+	static bool lowHPWarningActive;
 
 	static std::vector<int>remainingTimeActiveIndexes;
```
[the remark this answers is not in the document]
Comment (GBMiro) [reply Penteract/Assets/Scripts/HUDController.h:105]: Actually I think we don't longer need them to be static. If I recall correctly, we made them static to be able to call them from other scripts for the VS2 (this was before we had the GET_SCRIPT method).

If David agrees, we will make a branch to remove all unnecessary static variables and test everything keeps working

**Review by isaacfdez — APPROVED** [review]

**Comment by isaacfdez** [comment]
Have you talked with the design team? I think it could look better on both sides of the screen.

**Comment by GBMiro** [comment]
> Have you talked with the design team? I think it could look better on both sides of the screen.

Not directly but I was given this reference. The position and image used will change when building the final HUD but if they finally want to add this effect on both sides it won't be a problem (just a little tweak in the code and HUD structure)

LowHPDesign

**Review by devildrake — APPROVED** [review]

**Pull request merged**